### PR TITLE
Updating pubsub to match gax-java 0.0.12

### PR DESCRIPTION
**Pull request by michaelbausor**
Updated pubsub to latest generated surface
Updated DefaultPubSubRpc and PublisherApiTest to match new surface

### CODE REVIEW

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 1]
```diff
@@ -306,7 +324,7 @@ private Topic createTopic(Topic request) {
 
   // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
   /**
-   * Adds one or more messages to the topic. Generates `NOT_FOUND` if the topic
+   * Adds one or more messages to the topic. Returns `NOT_FOUND` if the topic
```
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: It looks like we are losing a documentation update that was made for pubsub. (We updated the wording of "Returns" and changed it to "Generates")

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 2]
```diff
@@ -133,11 +131,15 @@
...
 
+  public final PublisherSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate TOPIC_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225085/issues/source?bid=3123109&fileBranchId=3277889#l141)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 3]
```diff
@@ -133,11 +131,15 @@
       listTopicSubscriptionsPagedCallable;
   private final ApiCallable<DeleteTopicRequest, Empty> deleteTopicCallable;
 
+  public final PublisherSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225085/issues/source?bid=3123109&fileBranchId=3277889#l138)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 4]
```diff
@@ -138,11 +136,18 @@
   private final ApiCallable<PullRequest, PullResponse> pullCallable;
   private final ApiCallable<ModifyPushConfigRequest, Empty> modifyPushConfigCallable;
 
+  public final SubscriberSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225067/issues/source?bid=3123109&fileBranchId=3277889#l143)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 5]
```diff
@@ -138,11 +136,18 @@
...
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate SUBSCRIPTION_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225067/issues/source?bid=3123109&fileBranchId=3277889#l146)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 6]
```diff
@@ -138,11 +136,18 @@
   private final ApiCallable<PullRequest, PullResponse> pullCallable;
   private final ApiCallable<ModifyPushConfigRequest, Empty> modifyPushConfigCallable;
 
+  public final SubscriberSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225087/issues/source?bid=3123109&fileBranchId=3277889#l143)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 7]
```diff
@@ -138,11 +136,18 @@
...
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate SUBSCRIPTION_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225087/issues/source?bid=3123109&fileBranchId=3277889#l146)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 8]
```diff
@@ -133,11 +131,15 @@
...
 
+  public final PublisherSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate TOPIC_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225065/issues/source?bid=3123109&fileBranchId=3277889#l141)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 9]
```diff
@@ -133,11 +131,15 @@
       listTopicSubscriptionsPagedCallable;
   private final ApiCallable<DeleteTopicRequest, Empty> deleteTopicCallable;
 
+  public final PublisherSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2515225065/issues/source?bid=3123109&fileBranchId=3277889#l138)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 10]
```diff
@@ -378,9 +428,6 @@ public final Subscription getSubscription(String subscription) {
   /**
    * Gets the configuration details of a subscription.
    *
-   * If the topic of a subscription has been deleted, the subscription itself is
```
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: This is another doc update that is being lost.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 11]
```diff
@@ -65,10 +66,10 @@
  *
  * <pre>
  * <code>
- * try (PublisherApi publisherApi = PublisherApi.defaultInstance()) {
- *   // make calls here
- * String name = "";
- * Topic callResult = createTopic(name);
+ * try (PublisherApi publisherApi = PublisherApi.createWithDefaults()) {
+ *   String name = "";
+ *
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Why this empty line?

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 12]
```diff
@@ -66,13 +67,13 @@
...
- * String topic = "";
- * PushConfig pushConfig = PushConfig.newBuilder().build();
- * int ackDeadlineSeconds = 0;
- * Subscription callResult = createSubscription(name, topic, pushConfig, ackDeadlineSeconds);
+ * try (SubscriberApi subscriberApi = SubscriberApi.createWithDefaults()) {
+ *   String name = "";
+ *   String topic = "";
+ *   PushConfig pushConfig = PushConfig.newBuilder().build();
+ *   int ackDeadlineSeconds = 0;
+ *
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Also here why the empty line?

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 13]
```diff
@@ -280,6 +332,8 @@ public void close() throws IOException {
    * plus (`+`) or percent signs (`%`). It must be between 3 and 255 characters
    * in length, and it must not start with `"goog"`.
    * @param topic The name of the topic from which this subscription is receiving messages.
+   * The value of this field will be `_deleted-topic_` if the topic has been
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: This comment is not clear to me. What has this to do with `create`? And why was this removed from `get` and `list` methods?

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java [hunk 14]
```diff
@@ -69,12 +69,12 @@ public DefaultPubSubRpc(PubSubOptions options) throws IOException {
     try {
       // Provide (and use a common thread-pool).
       // This depends on https://github.com/googleapis/gax-java/issues/73
-      PublisherSettings.Builder pbuilder = PublisherSettings.defaultInstance().toBuilder();
+      PublisherSettings.Builder pbuilder = PublisherSettings.defaultBuilder();
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java]: Could also `pbuilder.provideChannelWith(credentials)` be used instead of
```
pbuilder.provideChannelWith(ConnectionSettings.newBuilder()
    .provideCredentialsWith(credentials).build());
```

**Comment by mziccard** [comment]
Also how is the status of https://github.com/googleapis/gax-java/issues/73 in gax 0.0.12?

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberSettings.java [hunk 15]
```diff
@@ -408,8 +451,14 @@ public Builder provideChannelWith(ConnectionSettings settings) {
     }
 
     @Override
-    public Builder setExecutor(ScheduledExecutorService executor) {
-      super.setExecutor(executor);
+    public Builder provideChannelWith(Credentials credentials) {
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberSettings.java]: If would be nice if there was a way of providing no credentials (e.g. for testing with a local emulator) to avoid having to pass a channel with `provideChannelWith`.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 16]
```diff
@@ -138,11 +135,18 @@
   private final ApiCallable<PullRequest, PullResponse> pullCallable;
   private final ApiCallable<ModifyPushConfigRequest, Empty> modifyPushConfigCallable;
 
+  public final SubscriberSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054014/issues/source?bid=3123109&fileBranchId=3277889#l142)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 17]
```diff
@@ -133,11 +130,15 @@
...
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054032/issues/source?bid=3123109&fileBranchId=3277889#l137)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 18]
```diff
@@ -138,11 +135,18 @@
...
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate SUBSCRIPTION_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054014/issues/source?bid=3123109&fileBranchId=3277889#l145)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 19]
```diff
@@ -133,11 +130,15 @@
...
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate TOPIC_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054032/issues/source?bid=3123109&fileBranchId=3277889#l140)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 20]
```diff
@@ -138,11 +135,18 @@
   private final ApiCallable<PullRequest, PullResponse> pullCallable;
   private final ApiCallable<ModifyPushConfigRequest, Empty> modifyPushConfigCallable;
 
+  public final SubscriberSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054034/issues/source?bid=3123109&fileBranchId=3277889#l142)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java [hunk 21]
```diff
@@ -138,11 +135,18 @@
...
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate SUBSCRIPTION_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054034/issues/source?bid=3123109&fileBranchId=3277889#l145)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 22]
```diff
@@ -133,11 +130,15 @@
       listTopicSubscriptionsPagedCallable;
   private final ApiCallable<DeleteTopicRequest, Empty> deleteTopicCallable;
 
+  public final PublisherSettings getSettings() {
+    return settings;
+  }
+
   private static final PathTemplate PROJECT_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054012/issues/source?bid=3123109&fileBranchId=3277889#l137)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java [hunk 23]
```diff
@@ -133,11 +130,15 @@
...
-      PathTemplate.create("projects/{project}");
+      PathTemplate.createWithoutUrlEncoding("projects/{project}");
 
   private static final PathTemplate TOPIC_PATH_TEMPLATE =
```
Comment (mziccard) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2516054012/issues/source?bid=3123109&fileBranchId=3277889#l140)

File: gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java (hunk 13 above)
Comment (michaelbausor) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java]: My mistake, I generated from the unedited proto. Fixed in the latest commit.

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java (hunk 14 above)
Comment (michaelbausor) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java]: Fixed in latest commit

**Comment by michaelbausor** [comment]
https://github.com/googleapis/gax-java/issues/73 fixed in https://github.com/googleapis/gax-java/pull/75
Closed issue now
